include/irr/core/memory/dynamic_array.h

```diff
@@ -0,0 +1,112 @@
+#ifndef __IRR_DYNAMIC_ARRAY_H_INCLUDED__
+#define __IRR_DYNAMIC_ARRAY_H_INCLUDED__
+
+#include "irr/macros.h"
+#include "irr/core/Types.h"//for core::allocator
+
+namespace irr
+{
+namespace core
+{
+
+template<typename T, class allocator = core::allocator<T> >
+class IRR_FORCE_EBO dynamic_array : public Uncopyable, public Unmovable
```

Comment:
@Crisspl can you read&review this code if it will function correctly?

Reply:
ah, i've seen this and forgot to comment. 
Indeed i forgot to make copy/move ctors/assignments for `dynamic_array`, but i see you've made it uncopyable and unmovable. Why is that?

Reply:
because the allocator now is used in the class-override of the `new` and `delete` operators, the allocator allocates enough space for the `base` members + invisible array of data after the `dynamic_array` object.

Thus making `sizeof(dynamic_array<T>)` a lie...

Hence it should not be possible to createa `dynamic_array<T>` on the stack, and two arrays of same type but different length are incompatible with each other (we cannot move nor copy the contents as the amount of space might not match).

So if you want to copy or trade contents, trade and copy pointers or references to `dynamic_array`.

Reply:
argh shit I just realized I made a mistake.